components/collective-page/hero/Hero.js

```diff
@@ -242,28 +244,22 @@ const Hero = ({ collective, host, isAdmin, onPrimaryColorChange }) => {
                       values={{
                         FiscalHost: <DefinedTerm term={Terms.FISCAL_HOST} color="black.700" />,
                         hostName: (
-                          <LinkCollective collective={host}>
-                            <Span data-cy="fiscalHostName" color="black.700">
-                              {host.name}
-                            </Span>
-                          </LinkCollective>
+                          <OverflowCollectiveName
+                            collective={host}
+                            dataCy="fiscalHostName"
+                            cutOff={parentIsHost && host.name.length >= 30 ? host.name.length : 30}
+                          />
```

Comment:
Simply using the [`truncate` function from `lodash`](https://docs-lodash.com/v4/truncate/) might be sufficient. 

```suggestion
                          <LinkCollective collective={host}>
                            <Span data-cy="fiscalHostName" color="black.700">
                              {truncate(host.name, {length: 30})}
                            </Span>
                          </LinkCollective>
```